src/ripple/rpc/impl/RPCHandler.cpp

```diff
@@ -157,6 +157,12 @@ error_code_i fillHandler (Context& context,
         return rpcNO_NETWORK;
     }
 
+    if (context.app.getOPs().isAmendmentBlocked() &&
+        handler->condition_ & NEEDS_CURRENT_LEDGER)
+    {
+        return rpcAMENDMENT_BLOCKED;
+    }
+
     if (!context.app.config().standalone() &&
         handler->condition_ & NEEDS_CURRENT_LEDGER)
```

Comment:
I like the idea of using these conditions to determine what commands to reject when an amendment blocked.
At first, I was concerned about `NO_CONDITION` commands such as `account_channels`, `account_objects`, `channel_authorize`, `channel_verify`, and `ledger_entry` that are affected by amendments. But commands specific to an amendment would simply return an `unknownCmd` error, and when I reverted the commit that added `PayChan`, rippled crashed before I could even try looking up account objects/channels or ledger entries
```
Terminating thread doJob: OrderBookDB::update: unhandled St13runtime_error 'invalid ledger entry type'
```

All that to say, there are already some limitations in rippled's ability to serve RPC requests while amendment blocked.